### PR TITLE
Development simple coredns 

### PR DESCRIPTION
**Pull request by BolaNasr**
can use coredns client :
```python
        #create etcd client
        cl = j.clients.coredns.configure(instance_name="main",host="10.144.72.95",password="njufdmrq3k")
        #create zones
        cl.zone_create('test.example.com','10.144.13.199',types='A')
        cl.zone_create('example.com','10.144.218.172',types='AAAA')
        #add records in etcd
        cl.add_records() 
        #get records from etcd
        cl.deploy()
        #remove records from etcd
        cl.remove_record()
```

### CODE REVIEW

File: JumpscaleLib/clients/coredns/CoreDNS.py [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
-import math
-from ipaddress import IPv4Address, IPv6Address
-from ipaddress import AddressValueError, NetmaskValueError
+from .encoding import *
+from enum import Enum
+JSConfigBase = j.tools.configmanager.JSBaseClassConfig
 
 TEMPLATE = """
-etcdpath = "skydns"
+etcdpath = "/hosts"
```
Comment (zaibon) [inline JumpscaleLib/clients/coredns/CoreDNS.py]: why should this be configurable ?

File: JumpscaleLib/clients/coredns/CoreDNS.py [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
+class RecordType(Enum):
+    A = "A"
+    AAAA = "AAAA"
+    SRV = "SRV"
+    TXT = "TXT"
+
 class ResourceRecord:
 
-    def __init__(self, name, type, ttl, rrdata,
+    def __init__(self, domain, rrdata, types=RecordType.A,  ttl=300,
```
Comment (zaibon) [inline JumpscaleLib/clients/coredns/CoreDNS.py]: why did you rename `type` to `types` ? You only accept a single value, not multiple

File: JumpscaleLib/clients/coredns/CoreDNS.py [hunk 3]
```diff
@@ -1,29 +1,32 @@
...
                              priority=None, port=None # SRV
                 ):
         """ DNS Resource Record.
 
             name    : Record Name.  Corresponds to first field in
                       DNS Bind Zone file entries.  REQUIRED.
-            type    : Record Type.  CNAME, A, AAAA, TXT, SRV.  REQUIRED
-            ttl     : time to live.  REQUIRED
+            types    : Record Type.  CNAME, A, AAAA, TXT, SRV.  REQUIRED
```
Comment (zaibon) [inline JumpscaleLib/clients/coredns/CoreDNS.py]:
```suggestion
            type    : Record Type.  CNAME, A, AAAA, TXT, SRV.  REQUIRED
```

File: JumpscaleLib/clients/coredns/CoreDNS.py [hunk 4]
```diff
@@ -1,29 +1,32 @@
...
-from ipaddress import AddressValueError, NetmaskValueError
+from .encoding import *
+from enum import Enum
+JSConfigBase = j.tools.configmanager.JSBaseClassConfig
 
 TEMPLATE = """
-etcdpath = "skydns"
+etcdpath = "/hosts"
+etcd_instance = "main"
 secrets_ = ""
```
Comment (zaibon) [inline JumpscaleLib/clients/coredns/CoreDNS.py]: is `secrets_`  used somewhere?

File: JumpscaleLib/clients/coredns/CoreDNS.py [hunk 5]
```diff
@@ -112,380 +111,28 @@ def secrets(self):
...
+            self._etcd_client = j.clients.etcd.get(self._etcd_instance)
+        return self._etcd_client
+
+    def zone_create(self, domain, rrdata, types=RecordType.A,  ttl=300,
+                             priority=None, port=None ):
+         
+        zone =ResourceRecord(domain, rrdata, types, ttl, priority, port)
+        self.zones.append(zone)
+
+    def add_records(self):
```
Comment (zaibon) [inline JumpscaleLib/clients/coredns/CoreDNS.py]: in the traefik SAL we use `deploy` as method name when we write data to etcd. I think it is a more explicit name then `add_records`

File: JumpscaleLib/clients/coredns/CoreDNS.py [hunk 6]
```diff
@@ -1,29 +1,32 @@
...
-
-"""
-
-from copy import deepcopy
+from Jumpscale import j
 import json
-import math
-from ipaddress import IPv4Address, IPv6Address
-from ipaddress import AddressValueError, NetmaskValueError
+from .encoding import *
```
Comment (zaibon) [inline JumpscaleLib/clients/coredns/CoreDNS.py]: don't use global import, this mask auto completion in some editors and I prefer to be explicit about those things in case in the future we had more stuff in the `encoding` module that are not needed here.

File: JumpscaleLib/clients/coredns/CoreDNS.py (hunk 4 above)
Comment (BolaNasr) [reply JumpscaleLib/clients/coredns/CoreDNS.py]: i think we use it future , there is just simple coredns client without sercret
